### PR TITLE
feat: initialize foundry repo for Swaplace+CCIP #142

### LINKED ISSUE #142: feat: initialize foundry repo for Swaplace-CCIP
## Feature Request

Initialize the Foundry repository in this [branch](https://github.com/blockful-io/swaplace-ccip/tree/feat/foundry)

### Describe the Feature Request

- Download Foundry in the repository
- Add OpenZeppelin to the remappings
- Migrate the contracts and scripts into their respective folders

### Describe Preferred Solution

Only setting up the repo and migrating the files is necessary for this issue, among setting OpenZeppelin and RPC configurations.

### Additional Context

I'll create a lot of issues on top of this resolution. CCIP will be a medium<>high difficulty level and will count for this incentives cycle. The sooner this is completed the faster we can release the issues.

### PR DESCRIPTION
**Pull request by 0xjoaovictor**
This PR close [#142](https://github.com/blockful-io/swaplace-contracts/issues/142)

Checklist:

- [x] Removed old files in the swaplace-ccip repository
- [x] Contracts brought from swaplace-contracts to swaplace-ccip
- [x] Removed the hardhat
- [x] Installed Foundry setup
- [x] Installed openzeppelin
- [x] Config the remappings.txt with openzeppelin path

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,4 +11,13 @@
 /cache
 
 # Typechains assists with type safety when interacting with contracts.
-/typechain-types
\ No newline at end of file
+/typechain-types
+
```
Comment (luislucena16) [inline .gitignore:16]: add this: 
`cache_forge`

File: foundry.toml
Comment (luislucena16) [inline foundry.toml:1]: maybe here you can use this:
```
# Default profile with all base configs
[profile.default]
src = 'src'
out = 'out'
libs = ['node_modules', 'lib']
cache_path  = 'cache_forge'
eth_rpc_url = 'https://mainnet.infura.io/v3/apiKey' // if we need this
fork-block-number = 18135100 // if we need this
optimizer = false
solc-version = "0.8.21"
```

**Comment by 0xneves** [comment]
Hey @0xjoaovpsantos, I'll report here the review directly because using the review feature I would have to report in a lot of files.

1- It was not part of the task to bring Swaplace contracts from swaplace-contracts, the ccip contracts were already in the specified branch requested in the issue
2- The libs are not supposed to be committed, they are just like the node_modules folder from hardhat and should be left for the user to install after cloning the repository
3- You completely deleted the core CCIP.sol contracts (🙈🤣) that were already in the feat/foundry repo, which are the ones we need.
4- You also deleted the CCIP scripts which must stay in the scripts folder to aid users understand the CCIP flux to correctly implement the tests in Solidity.

Overall I think you misunderstood the entire issue and maybe lacked knowledge about what CCIP is.
Just to summarize. Swaplace-contracts is one thing, swaplace-ccip is another different thing, just like swaplace-solana or swaplace-cairo. 
Swaplace + hardhat is the core multichain EVM contract set while swaplace + foundry is the core cross-chain EVM contract set.

Let me know if you want to try again, our deadline is Thursday. After that, this will become a hot task worthing 2x points to the first one to open the correct PR.

**Review by 0xneves — CHANGES_REQUESTED** [review]
Requesting for a remake based on my last comment above.

**Comment by 0xjoaovictor** [comment]
OH NO SORRY MAN @0xneves 

I'll fix all points now!!!

**Comment by 0xjoaovictor** [comment]
@0xneves sorry man!! I fixed the points:

[x] 1- It was not part of the task to bring Swaplace contracts from swaplace-contracts, the ccip contracts were already in the specified branch requested in the issue
[x] 2- The libs are not supposed to be committed, they are just like the node_modules folder from hardhat and should be left for the user to install after cloning the repository
[x] 3- You completely deleted the core CCIP.sol contracts (🙈🤣) that were already in the feat/foundry repo, which are the ones we need.
[x] 4- You also deleted the CCIP scripts which must stay in the scripts folder to aid users understand the CCIP flux to correctly implement the tests in Solidity.

**Review by luislucena16 — COMMENTED** [review]
in general I think you should delete the `lib` folder and bring back the `CCIP` contracts and scripts that were deleted

**Comment by 0xjoaovictor** [comment]
> in general I think you should delete the `lib` folder and bring back the `CCIP` contracts and scripts that were deleted

All CCIP Contracts and Scripts already are in the main branch :)

**Comment by luislucena16** [comment]
yes, my comments were late. I hadn't posted them

**Comment by 0xneves** [comment]
The current folder structure is incorrect and Foundry will not compile or work properly. This issue is still blocking the creation of more tasks regarding CCIP. Please help me unstuck the opportunities!

**Review by 0xneves — CHANGES_REQUESTED** [review]
The current folder structure is incorrect and Foundry will not compile or work properly. This issue is still blocking the creation of more tasks regarding CCIP. Please help me unstuck the opportunities!

**Comment by 0xjoaovictor** [comment]
Hey @0xneves how are you?
I fixed it!!! 

- [x] Changed the root of project to 'src' foler
- [x] Config remapping.txt with chainlink and oepnzeppelin packages

`forge build` in the root of project:

image

**Comment by 0xneves** [comment]
> Hey @0xneves how are you? I fixed it!!!
> 
> * [x]  Changed the root of project to 'src' foler
> * [x]  Config remapping.txt with chainlink and oepnzeppelin packages
> 
> `forge build` in the root of project:
> 
> image

Perfect man!! The first foundry setups we never forget hahaha
Thank you for reorganizing the repo and sorry for the late response, we've been channeling the hadron colliders to the dApp repo 🤣

**Review by 0xneves — COMMENTED** [review]
Perfect, approved this PR and merged to main

**Pull request merged**